### ISSUE TITLE
GLideN64 displays black screen/nothing on my laptop (IntelHDGraphics)

### ISSUE THREAD

**Issue by TheArcadeStriker** [gonetz/GLideN64#1831]
My laptop has a Intel Pentium B960 and Intel HD Graphics (Sandy Bridge IIRC), and I have tested the latest version of the plugin in PJ64, m64p (the one included) and Bizhawk; but the result always is a black screen. Now, I have heard that it requires OGL3.X, but my laptop DOES support OpenGL 3.1 (but not 3.3 unless I'm on Linux, which I don't use) so I'm not sure why it refuses to work properly.

**Comment by loganmc10** [comment]
It requires OGL 3.3+

**Comment by gonetz** [comment]
Yes,  OGL 3.3+ is must. I think, you should have error messages in gliden64.log

**Comment by TheArcadeStriker** [comment]
[gliden64.log](https://github.com/gonetz/GLideN64/files/2068663/gliden64.log)
Here's my log file.

**Comment by TheArcadeStriker** [comment]
I guess it's another one to add to the "3.3 no-go" list :l
But I can understand that newer versions add support for better features that could be essential in getting it working. Anyways, I'm sure there were older versions that DID work in 3.1; does anyone have a idea about where I can get the last 3.1 compatible version?

**Comment by gonetz** [comment]
@TheArcadeStriker 
> Here's my log file.

It contains lots of `'sampler2DMS' :  precision statement not allowed for type` errors.
No shaders created and you have black screen.
Disable multi-sampling to avoid this error. 

BTW, @loganmc10, do you know why sampler2DMS has precision qualifier? I can't remember why it is there. Samplers of sampler2D type have no precision in the code.

**Comment by loganmc10** [comment]
I don't think that was me, but I assume it's probably because GLES required it?

Even if that is fixed, I'm pretty sure there are lots of other things that are needed that 3.1 doesn't provide, off the top of my head there is ARB_depth_clamp, but there are probably a few other very significant things in the shaders that won't be supported

**Comment by gonetz** [comment]
> I don't think that was me

Git blame points on me, but I don't remember why I added that qualifier. I also assumed that it can be GLES related.

**Comment by TheArcadeStriker** [comment]
![image](https://user-images.githubusercontent.com/24653951/41064602-093712d2-69aa-11e8-8cb9-8d4758b8c892.png)
Can't find any multisampling in the options. But I guess that if the plugin needs 3.3, then at least I guess it could be useful to have any old version that supports 3.1, or a 3.1 compatible version.

**Comment by gonetz** [comment]
According to settings on your configuration panel, you have Anti-aliasing (MSAA) disabled.
In that case you must not have sampler2DMS in your shaders. May be it is an old log?
Please remove gliden64.log file, reset GLideN64 settings to defaults  and run the emulator again.

Regarding old versions: you may try to run Public Release 1:
https://github.com/gonetz/GLideN64/releases/tag/Public_Release_1.2
It uses OpenGL compatibility profile. It is oldest version.

You still may use Glide64 plugin, as it is OpenGL 2.X compatible.
